### DISCUSSION TITLE
How to load subRows asynchronously?

### DISCUSSION THREAD

**Discussion by hwangar** [TanStack/table#4541] [General]
The use case is: load entities and expand subentities lazily... ATM getSubRows is fetched synchronously, so no way to implement a server fetching around subrows

**Comment by Chrisdo82** [comment]
Did you find a solution? I'm struggling to implement the same use case. Apart from setting the `manualExpanding` option to true, the documentation or examples are not very helpful in this case. My data model only has the information if subrows are present or not, if yes, then the subrows should be reloaded asynchronously when expanding. 

What is the best way to achieve this with react-table?

**Comment by vuledang-ncc** [comment]
I'm getting the same issue.

**Comment by dilfin07** [comment]
one from solution is to recursively update the data array, with flags `autoResetExpanded: false `

**Comment by steveariss** [comment]
Hoping to implement the same.

I'd like subrow data to only be loaded if the parent row is expanded.

**Comment by lukasbach** [comment]
I found that `getSubRows` is called again when the data props updates. So for me it looks like it works to force the data field to update every time data used for subrows changes, which isn't nice but seems to be working.

**Comment by sanchitos** [comment]
Guys I was suffering too until I checked previous version's examples.

In case you are helpless like I was: [https://github.com/TanStack/table/blob/main/examples/react/sub-components/src/main.tsx](https://github.com/TanStack/table/blob/main/examples/react/sub-components/src/main.tsx)
